webrecorder/browsertrix
# Update collection archived item lists

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1406



### Changes
New features & enhancements:
- New UI for collection item selection dialog
- Consistent data table styles for collection list and collection item list

Refactors:
- Adds `btrix-table` as low-level table component
- Adds `btrix-archived-item-list`, removes `checkbox-list` and deprecates `crawl-list`
- Upgrades Shoelace for `sl-tree` fixes
- Fixes `ArchivedItem` typing


### Manual testing
Prerequisite: run `yarn` to get upgraded shoelace version
1. Log in as crawler or admin user
2. Go to Collections. Verify list renders as expected
3. Click collection with items and then "Archived Items". Verify list renders as expected with new "Pages Crawled" column"
4. Click "Select Items". Verify crawl workflows render with correct selection
5. Toggle "Auto-Add". Verify setting saves on toggle
6. Filter and sort items. Verify list view updates as expected
7. Update selection and click "Save Selection". Verify items as expected

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Collection Detail - Archived Items | <img width="1128" alt="Screenshot 2024-01-17 at 1 15 29 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/d2b4ccd8-7094-46f6-ad13-cc46688ecfac"> |
| Collection Detail - Select Items Dialog - Crawls | <img width="1144" alt="Screenshot 2024-01-17 at 1 15 42 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/767aaa3c-2713-4de2-8a2c-99598ef5e840"> |
| Collection Detail - Select Items Dialog - Crawls (Only in collection) | <img width="1121" alt="Screenshot 2024-01-17 at 1 15 58 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/53e7a2fd-45d2-4e90-a941-cc1c5005e1f8"> |
| Collection Detail - Select Items Dialog - Uploads | <img width="1121" alt="Screenshot 2024-01-17 at 2 18 02 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/cb0ccfbd-b376-4949-9144-74a03b3ffba1"> |

### Follow-ups
Further refactors outside of the collections feature will be done in fast-follows tracked here:
- https://github.com/webrecorder/browsertrix-cloud/issues/1471
- https://github.com/webrecorder/browsertrix-cloud/issues/1483
